src/Language/PureScript/CST/Parser.y

```diff
@@ -188,12 +193,14 @@ qualIdent :: { QualifiedName Ident }
   | 'as' {% toQualifiedName Ident $1 }
   | 'hiding' {% toQualifiedName Ident $1 }
   | 'kind' {% toQualifiedName Ident $1 }
+  | 'role' {% toQualifiedName Ident $1 }
 
 ident :: { Name Ident }
   : LOWER {% toName Ident $1 }
   | 'as' {% toName Ident $1 }
   | 'hiding' {% toName Ident $1 }
   | 'kind' {% toName Ident $1 }
+  | 'role' {% toName Ident $1 }
```

Comment:
You need to add the other contextual keywords to `ident` and `qualIdent` as well (phantom, nominal, representational).